api/s2n.h

```diff
@@ -3324,6 +3326,19 @@ S2N_API extern const char *s2n_connection_get_kem_name(struct s2n_connection *co
  */
 S2N_API extern const char *s2n_connection_get_kem_group_name(struct s2n_connection *conn);
 
+/**
+ * Function to get the human readable key exchange group name for the connection, for example: 
+ * `secp521r1` or `SecP256r1MLKEM768`.
+ *
+ * @note This function replaces `s2n_connection_get_curve` and `s2n_connection_get_kem_group_name`, returning
+ * the named group regardless if a hybrid PQ group was negotiated or not. 
+ *
+ * @param conn A pointer to the s2n connection
+ * @param group_name A pointer that will be set to point to a const char* containing the group name
+ * @returns S2N_SUCCESS on success, S2N_FAILURE otherwise. `group_name` will be set on success.
+ */
+S2N_API extern int s2n_connection_get_key_exchange_group_name(struct s2n_connection *conn, const char **group_name);
```

Comment:
Were you planning on adding this API to the rust bindings in another PR?

Reply:
I technically don't need it for https://github.com/aws/s2n-quic/issues/2523 since s2n-quic uses the low level bindings, but I can add it as part of this PR for completeness

Reply:
Definitely not required then. But I always end up cursing myself because I end up having to do an extra PR later on. In particular, since you're deprecating an API here, it's a little weird to have that be inconsistent: https://docs.rs/s2n-tls/latest/s2n_tls/connection/struct.Connection.html#method.kem_name

Reply:
I've added the bindings